File: Week 6 - Python/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # User input
    word = get_string("Text: ")

    # Assigning variables to functions
    a = count_letters(word)
    b = count_words(word)
    c = count_sentences(word)

    # Computing C-L index
    L = (a / b) * 100
    S = (c / b) * 100

    index = (0.0588 * L) - (0.296 * S) - 15.8

    # Deciding what to print to terminal
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        # Round function applies in Python just the same as C
        print(f"Grade {round(index)}")


# Words function
def count_words(text):
    count = 0

    n = len(text)

    for i in range(n):
        if text[i].isspace():
            count += 1
        elif i == (n - 1):
            count += 1

    return count


# Letters function
def count_letters(text):
    count = 0

    for i in range(len(text)):
        if text[i].isalpha():
            count += 1

    return count


# Sentences function
def count_sentences(text):
    count = 0
    n = len(text)

    for i in range(n):
        if text[i] == '.':
            count += 1
        elif text[i] == '!':
            count += 1
        elif text[i] == '?':
            count += 1

    return count


main()

File: Week 2 - Arrays/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get user text
    string word = get_string("Text: ");

    // Assigning variables to the functions so I can easily refer to them
    float a = count_letters(word);
    float b = count_words(word);
    float c = count_sentences(word);

    float L = (a / b) * 100;
    float S = (c / b) * 100;

    // Computing the Coleman-Liau index
    float index = (0.0588 * L) - (0.296 * S) - 15.8;


    // Computing what to print to terminal based on result above
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        int h = round(index); // Played around with double and float to get the right answer. But very cleverly just used variable 'h'!
        printf("Grade %i\n", h);
    }

}


// Count number of letters in string and print as integer
int count_letters(string text)
{
    int count = 0;

    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        if (isalpha(text[i]))
        {
            count += 1;
        }
    }
    return count;

}

// Count words in the string using isspace to spot 'spaces'
int count_words(string text)
{
    int count = 0;

    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        if (isspace(text[i]))
        {
            count += 1;
        }
        else if (i == n)
        {
            count += 1;
        }

    }
    return count;
}

// Count sentences using ., ! and ? to denote end of a sentence.
// Did really well to create these functions from a mix a googling and lecture tips. Well done!
int count_sentences(string text)
{
    int count = 0;

    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        char a = text[i];

        if (a == '.')
        {
            count += 1;
        }
        else if (a == '!')
        {
            count += 1;
        }
        else if (a == '?')
        {
            count += 1;
        }
    }
    return count;

}
